src/script/engine.zig

```diff
@@ -358,6 +363,82 @@ pub const Engine = struct {
         try self.stack.pushByteArray(value);
     }
 
+    /// OP_NIP: Removes the second-to-top stack item
+    ///
+    /// # Returns
+    /// /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opNip(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const top_value = try self.stack.pop();
+        const second_to_top_value = try self.stack.pop();
+        try self.stack.pushByteArray(top_value);
+        defer self.allocator.free(top_value);
+        defer self.allocator.free(second_to_top_value);
+    }
+
+    /// OP_OVER: Copies the second-to-top stack item to the top
+    ///
+    /// /// # Returns
+    /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opOver(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const value = try self.stack.peek(1);
+        try self.stack.pushByteArray(value);
+    }
+
+    /// OP_SWAP: The top two items on the stack are swapped.
+    ///
+    /// /// # Returns
+    /// - "EngineError.StackUnderflow": if initial stack length < 2
+    fn opSwap(self: *Engine) !void {
+        if (self.stack.len() < 2) {
+            return error.StackUnderflow;
+        }
+        const top_value = try self.stack.pop();
+        const second_to_top_value = try self.stack.pop();
+        try self.stack.pushByteArray(top_value);
+        try self.stack.pushByteArray(second_to_top_value);
+        defer self.allocator.free(top_value);
+        defer self.allocator.free(second_to_top_value);
```

Comment:
Please find a way to take an item out of the stack and then push it back without having to allocator.dupe & allocator.free

Reply:
we implemented swap in the same way for [Shinigami](https://github.com/keep-starknet-strange/shinigami/blob/688e464c6bb394608a37dcc1cf0a9f8db2ee9f5f/src/opcodes/stack.cairo#L34) as well.